internal/controller/openstackimage_controller.go

```diff
@@ -147,16 +147,22 @@ func (r *OpenStackImageReconciler) reconcile(ctx context.Context, imageClient *g
 		logger.Info("resouce exists in OpenStack")
 	} else {
 		var err error
+
+		var imageVisibility *images.ImageVisibility
+		if resource.Spec.Visibility != "default" {
+			v := images.ImageVisibility(resource.Spec.Visibility)
+			imageVisibility = &v
```

Comment:
Since `Visibility` will now be a pointer:
```suggestion
		if resource.Spec.Visibility != nil {
			v := images.ImageVisibility(*resource.Spec.Visibility)
			imageVisibility = &v